### ISSUE TITLE
To provide the prediction probabilities for Binary classification task using the sigmoid activation in the output layer.

### ISSUE THREAD

**Issue by Mahima-ai** [tensorflow/tensorflow#45701]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.3
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
Feature: For binary classification with sigmoid activation in the output layer, the model.predict() should output the probability values for both class 0 and class 1 as it does in the case of multi-class classification.

Current behavior/state: Using the model.predict() method, the output is a numpy array with single probability value. If the probability value>0.5 then it belongs to class 1 else class 0. i.e. It does not provide the probability values for each class separately as in a multiclass classification which uses the softmax activation in the output layer.

**Will this change the current api? How?**
Minor change for the output of model.predict() for binary classification task with 1 neuron and sigmoid activation in the output layer.

**Who will benefit with this feature?**
This feature will be helpful for explaining the model. There are XAI libraries like Lime, Shap which depends on the probability values of each class for explaining the model. They take the model as an input parameter and expect the prediction/output of the model prediction in the form of probabilities for each class to explain the model.

**Any Other info.**
Lime takes a function as an input to its explainer. This function can be user-defined which can implicitly call TF model.predict() and output the probability values for both the classes based on the 0.5 threshold value. 
But in case of Shap, the Shap Deep Explainer (especially designed for TF and Pytorch Neural Netoworks) takes the TF/Pytorch model as input and internally calls the model.predict() and generates the shap values. Since, here in case of binary classification with sigmoid activation only single probability value is generated, so shap also generate single shap values instead of shap values for both the classes.
[Link to Lime](https://github.com/marcotcr/lime)
[Link to Shap Deep Explainer](https://shap.readthedocs.io/en/latest/generated/shap.DeepExplainer.html#shap.DeepExplainer)

**Comment by fchollet** [comment]
This is intended behavior: `model.predict()` simply runs the forward pass of the model on some input and returns the result. For a model that ends with a single output unit and a sigmoid activation, only scalar values are returned. `model.predict()` is different from sklearn's `predict`.
